data/src/main/java/com/microsoft/azure/kusto/data/exceptions/ExceptionUtils.java

```diff
@@ -0,0 +1,26 @@
+package com.microsoft.azure.kusto.data.exceptions;
+
+import com.microsoft.azure.kusto.data.Utils;
+
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.net.URL;
+
+public class ExceptionUtils {
+    public static DataServiceException createExceptionOnPost(Exception e, URL url, String kind) {
+        boolean permanent = false;
+        String prefix = "";
+        if (e instanceof IOException) {
```

Comment:
I think if you already change the exception message, you might as well make a proper subclass for IODataServiceException , so the users could filter on them too